### ISSUE TITLE
Kallax gen isn't picking up any models

### ISSUE THREAD

**Issue by peppage** [src-d/go-kallax#129]
$ go version
go version go1.8 windows/amd64

$ kallax -v
kallax version 1.0.0

When I run `$ kallax gen` it creates a generic kallax.go file. I'm running it in the tests dir of this project.
```go
// IMPORTANT! This is auto generated code by https://github.com/src-d/go-kallax
// Please, do not touch the code below, and if you do, do it under your own
// risk. Take into account that all the code you write here will be completely
// erased from earth the next time you generate the kallax models.
package tests

import (
        "fmt"

        "gopkg.in/src-d/go-kallax.v1/types"
)

var _ types.SQLType
var _ fmt.Formatter

type schema struct {
}

var Schema = &schema{}
```

I did run go get -u gopkg.in/src-d/go-kallax.v1/... and I did not get v1.1.0.

**Comment by erizocosmico** [comment]
Thanks for the report!

Should be updated now if you run again `go get -u`.

I see you're testing this with windows, so my money's on that. We have only tried this library on macOS and linux so far. I don't have access to any windows machine right now, but will try to setup a build with appveyor or something to investigate if it's just a windows issue or there's something else going on.

**Comment by peppage** [comment]
I am on the latest version now but same result.

Maybe something with the windows filesystem.

**Comment by JaCoB1123** [comment]
I'm having the same Problem on Windows using `kallax version 1.1.0`

**Comment by JaCoB1123** [comment]
I just added some debug output: `processModel` is returning nil.
Will continue investigating.

**Comment by erizocosmico** [comment]
Oops, closed this by accident. We're busy until the mid of this week in a few conferences, but we'll be looking into this more carefully after that.

@JaCoB1123 is it picking the structs, though? If it's picking the structs but failing in `processModel`, it may be related to https://github.com/src-d/go-kallax/pull/133, which I just merged and will be released in a few moments.

**Comment by JaCoB1123** [comment]
Yep, my PR should have fixed this issue.
The problem was simply, that the comparison to the BaseType did not work, because the path separators were inconsistent:
https://github.com/src-d/go-kallax/blob/master/generator/processor.go#L241

**Comment by erizocosmico** [comment]
Can you confirm @peppage so we can close this?

**Comment by peppage** [comment]
Yes this is def grabbing the models now.

running Kallax gen in /tests gives an error now but that might be a different issue. It's printing everything to stdout and ending with this

kallax: No file generated due to an occurred error:
kallax.go:5392:57: missing ',' in parameter list (and 17 more errors)

**Comment by erizocosmico** [comment]
Can you send me the specific full error output in a gist or something?

**Comment by peppage** [comment]
sure! https://gist.github.com/peppage/f20a4ad7d95a274e18331f84963a1625

**Comment by Serabe** [comment]
This line is generating the wrong type:

```go
func (q *QueryFixtureQuery) FindByAliasStringParam(v E:\workspace\gopath\src\gopkg.in\src-d\go-kallax.v1\tests\fixtures.AliasString) *QueryFixtureQuery
```

**Comment by Serabe** [comment]
I need a Windows machine to debug this. For removing the GoPath we use the following code:

```go
func removeGoPath(path string) string {
	path = filepath.ToSlash(path)
	for _, p := range parseutil.DefaultGoPath {
		p = filepath.ToSlash(p + "/src/")
		if strings.HasPrefix(path, p) {
			return strings.Replace(path, p, "", -1)
		}
	}
	return path
}
```

And that is not removing the GoPath because of some reason. @erizocosmico this function is a good candidate to extract to `go-parse-utils` since it is used in both kallax and proteus (or something similar).

**Comment by erizocosmico** [comment]
@Serabe I just tested this (https://github.com/src-d/go-kallax/pull/142) and it seems to work correctly on Windows (made some changes so is testable cross-platform, but did not really change anything). @peppage are you using version >= 1.1.2?

**Comment by erizocosmico** [comment]
Closing, tests say this is fixed. If it's not, ping us and we'll reopen.

**Comment by peppage** [comment]
Sorry for the delay, this is fixed! Ended up testing with version 1.2.4

**Comment by erizocosmico** [comment]
Thanks!
On Wed, 17 May 2017 at 17:06, Michael Peppler <notifications@github.com<mailto:notifications@github.com>> wrote:

Sorry for the delay, this is fixed! Ended up testing with version 1.2.4
